api4/team.go

```diff
@@ -103,6 +105,24 @@ func getTeamsForUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getTeamsUnreadForUser(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	// optional team id to be excluded from the result
+	teamID := r.URL.Query().Get("id")
+
+	unreadTeamsList, err := app.GetTeamsUnreadForUser(teamID, c.Session.UserId)
```

Comment:
You need to pass in `c.Params.UserId` (this is the URL param) not `c.Session.UserId` (this is the logged in user)